bookkeeper-server/src/main/java/org/apache/bookkeeper/client/DefaultEnsemblePlacementPolicy.java

```diff
@@ -51,18 +66,43 @@
             return newBookies;
         }
         List<BookieSocketAddress> allBookies;
-        synchronized (this) {
+        rwLock.readLock().lock();
+        try {
             allBookies = new ArrayList<BookieSocketAddress>(knownBookies);
+        } finally {
+            rwLock.readLock().unlock();
```

Comment:
Thats should be ok because if we end up choosing a bookie that just went down, the write will fail and we'll come back and change the ensemble. BTW, this could happen even without my change. So I'm not changing the situation any way.